File: LeetCode/퀴즈 3-2 연속 부분배열 합.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {
        unordered_map<int, int> table{{0, -1}};
        int mod = 0;
        for (int i = 0; i < nums.size(); i++) {
            mod = (mod + nums[i]) % k;
            if (!table.contains(mod)) {
                table[mod] = i;
            } else if (i - table[mod] > 1) {
                return true;
            }
        }
        return false;
    }
};

File: LeetCode/퀴즈 3-2 연속 부분배열 합.py
class Solution:
    def checkSubarraySum(self, nums: List[int], k: int) -> bool:
        table = {0: -1}
        mod = 0
        for i, x in enumerate(nums):
            mod = (mod + x) % k
            if mod not in table:
                table[mod] = i
            elif i - table[mod] > 1:
                return True
        return False